api/src/services/application-exporter.service.ts

```diff
@@ -87,6 +88,71 @@ export class ApplicationExporterService {
     return await zipExport(readStream, zipFilename, filename, isSpreadsheet);
   }
 
+  /**
+   *
+   * @param req
+   * @param queryParams
+   * @param isLottery a boolean indicating if the export is a lottery
+   * @param isSpreadsheet a boolean indicating if the export is a spreadsheet
+   * @returns a promise containing a secure download url
+   */
+  async exporterSecure<QueryParams extends ApplicationCsvQueryParams>(
+    req: ExpressRequest,
+    queryParams: QueryParams,
+    isLottery: boolean,
+    isSpreadsheet: boolean,
+  ): Promise<string> {
+    const user = mapTo(User, req['user']);
+    await this.authorizeExport(user, queryParams.id);
+
+    let filename: string;
+    let readStream: ReadStream;
+    let zipFilename: string;
+    const now = new Date();
+    const dateString = dayjs(now).format('YYYY-MM-DD_HH-mm');
+    if (isLottery) {
+      readStream = await this.spreadsheetExport(queryParams, user.id, true);
+      zipFilename = `listing-${queryParams.id}-lottery-${
+        user.id
+      }-${now.getTime()}`;
+      filename = `lottery-${queryParams.id}-${dateString}`;
+    } else {
+      if (isSpreadsheet) {
+        readStream = await this.spreadsheetExport(queryParams, user.id, false);
+      } else {
+        readStream = await this.csvExport(queryParams, user.id);
+      }
+      zipFilename = `listing-${queryParams.id}-applications-${
+        user.id
+      }-${now.getTime()}`;
+      filename = `applications-${queryParams.id}-${dateString}`;
+    }
+
+    const path = await zipExportSecure(
+      readStream,
+      zipFilename,
+      filename,
+      isSpreadsheet,
+    );
+
+    await uploadToS3(
```

Comment:
nit: As discussed offline, if you are missing any of these environment variables it throws an error on the API call. I would recommend adding a check to make sure the variables exist and throw an error if it doesn't.